src/test/java/DirectionsServletTest.java

```diff
@@ -0,0 +1,175 @@
+  private static final String MONTREAL_TO_MONTREAL_WITH_WAYPOINTS_JSON =
+      "[\"[DirectionsLeg: \"Montreal, QC, Canada\" -\u003e \"Kitchener, ON, Canada\" (45.50171230,-73.56721840 -\u003e 43.45184990,-80.49313410), duration\u003d6 hours 10 mins, distance\u003d632 km: 16 steps]\",\"[DirectionsLeg: \"Kitchener, ON, Canada\" -\u003e \"Waterloo, ON, Canada\" (43.45184990,-80.49313410 -\u003e 43.46430180,-80.52042120), duration\u003d8 mins, distance\u003d3.0 km: 5 steps]\",\"[DirectionsLeg: \"Waterloo, ON, Canada\" -\u003e \"Windsor, ON, Canada\" (43.46430180,-80.52042120 -\u003e 42.31486680,-83.03656800), duration\u003d2 hours 55 mins, distance\u003d291 km: 18 steps]\",\"[DirectionsLeg: \"Windsor, ON, Canada\" -\u003e \"Montreal, QC, Canada\" (42.31486680,-83.03656800 -\u003e 45.50171230,-73.56721840), duration\u003d8 hours 42 mins, distance\u003d897 km: 21 steps]\"]";
+  private static final String MONTREAL_TO_WATERLOO_NO_WAYPOINTS_JSON =
+      "[\"[DirectionsLeg: \"Montreal, QC, Canada\" -\u003e \"Waterloo, ON, Canada\" (45.50171230,-73.56721840 -\u003e 43.46430180,-80.52042120), duration\u003d6 hours 11 mins, distance\u003d638 km: 22 steps]\"]";
+
+  @BeforeClass
+  public static void classInit() throws GeneralSecurityException, IOException {
+    when(directionsClientFactory.getDirectionsClient(Mockito.any())).thenReturn(directionsClient);
+  }
+
+  @Before
+  public void init() throws IOException {
+    request = Mockito.mock(HttpServletRequest.class);
+    response = Mockito.mock(HttpServletResponse.class);
+
+    // Writer used in get/post requests to capture HTTP response values
+    stringWriter = new StringWriter();
+    printWriter = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(printWriter);
+  }
+
+  @After
+  public void clear() {
+    // Dump contents after each test
+    stringWriter.getBuffer().setLength(0);
+  }
+
+  @Test
+  public void windsorToMontrealWithWaypoints() {
+    // Get optimized route of travel from Windsor, ON to Montreal, QC with waypoints
+    // Markham, ON,
+    // Waterloo, ON and Quebec City, QC in between.
+    // Should return four legs: Windsor, ON -> Waterloo, ON -> Markham, ON -> Quebec
+    // City, QC ->
+    // Montreal, QC.
+    try {
+      when(directionsClient.getDirections(
+              Mockito.anyString(), Mockito.anyString(), Mockito.any(String[].class)))
+          .thenReturn(windsorToMontrealWithWaypoints);
+    } catch (ApiException | InterruptedException | IOException e) {
+      e.printStackTrace();
+    }
```

Comment:
I'm not completely sure what's the best way to mock methods that throw checked exceptions. I think you could either just re-throw
```
catch(Exception e) {
   throw e;
}
```
Or add it to the test declaration, i.e., `public void windsorToMontrealWithWaypoints() throws <exception>`
(it might be worth checking on Google what's the standard way)

Reply:
I believe you're right about that, it seems like adding exceptions to the test declaration without having to rethrow is [sufficient](https://stackoverflow.com/questions/54514021/mockito-mock-method-throws-exception). Thanks!

Reply:
`RESOLVED`